src/main/java/com/google/codeu/servlets/FormHandlerServlet.java

```diff
@@ -26,6 +33,31 @@
 @WebServlet("/my-form-handler")
 public class FormHandlerServlet extends HttpServlet {
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    ServletOutputStream out = response.getOutputStream();
+    out.println("<h1>Here is the post you uploaded</h1>");
+    out.println("<ul>");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
What are these next few lines for? (43-54)

Reply:
This is similar to the code that you had before. Instead of getting it directly from the user, as was done previously, this code accesses the data from the datastore. The doPost() method adds the data to the datastore. So when a user uploads a photo, it goes to the datastore. Then the doGet() method grabs that data and posts it!

Reply:
Just realized there's an extra line, I'll fix that!